plugins/modules/discovery_intent.py

```diff
@@ -1338,17 +1343,62 @@ def get_task_status(self, task_id=None):
                 self.log(msg, "CRITICAL")
                 self.module.fail_json(msg=msg)
                 return False
+
             self.log("Task status for the task id (before checking status) {0} is {1}".format(str(task_id), str(response)), "INFO")
             progress = response.get('progress')
-            if progress in ('In Progress', 'Inventory service initiating discovery'):
+            try:
+                progress_value = int(progress)
+                result = True
+                self.log("The Process is completed", "INFO")
```

Comment:
Can we add more information here? 
Discovery process is completed ... something more meaningful?